custom_linting_rules/noExternalHttpLinkRule.ts

```diff
@@ -0,0 +1,99 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+import { ErrorTolerantWalker } from '../node_modules/tslint-microsoft-contrib/utils/ErrorTolerantWalker';
+import { ExtendedMetadata } from '../node_modules/tslint-microsoft-contrib/utils/ExtendedMetadata';
+import { Utils } from '../node_modules/tslint-microsoft-contrib/utils/Utils';
+
+import {
+  getJsxAttributesFromJsxElement,
+  getStringLiteral,
+  isEmpty
+} from '../node_modules/tslint-microsoft-contrib/utils/JsxAttribute';
+
+const FAILURE_STRING: string = 'Anchor tags with an external link must use https';
+
+/**
+ * Implementation of the no-external-http-link rule.
+ */
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: ExtendedMetadata = {
+    ruleName: 'tno-external-http-link',
+    type: 'functionality',
+    description: 'Anchor tags with an external link must use https',
+    options: null,
+    optionsDescription: '',
+    typescriptOnly: true,
+    issueClass: 'SDL',
+    issueType: 'Error',
+    severity: 'Critical',
+    level: 'Mandatory',
+    group: 'Security',
+    commonWeaknessEnumeration: '242,676'
```

Comment:
Do we know what this extended metadata is for and what it means? I tried looking this up, but wasn't sure what it meant.

Reply:
Again following the formatting of the microsoft tslint lib, but I don't think using extendedMetadata is necessary. 

I think its necessary to define rule metadata, but we could use the standard tslint metadata interface, like [this](https://github.com/palantir/tslint-react/blob/master/src/rules/jsxAlignmentRule.ts#L24)